.github/workflows/publish-image.yml

```diff
@@ -0,0 +1,49 @@
+name: Docker Image Build and Publish
+
+on:
+  push:
+    branches: [ "main" ]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+
+    - name: Checkout repository
+      uses: actions/checkout@v4
```

Comment:
Please use hashes instead of SemVers for all GitHub actions references. Otherwise, this looks great!

E.g.
```
uses: actions/checkout@dslkjfn234nb1123 //v4
```

Reply:
Done